exporter/clickhouseexporter/exporter_logs.go

```diff
@@ -178,31 +178,28 @@ SETTINGS index_granularity=8192, ttl_only_drop_parts = 1;
                                   )`
 )
 
-var driverName = "clickhouse" // for testing
-
 // newClickhouseClient create a clickhouse client.
 func newClickhouseClient(cfg *Config) (*sql.DB, error) {
-	dsn, err := cfg.buildDSN(cfg.Database)
+	db, err := cfg.buildDB(cfg.Database)
 	if err != nil {
 		return nil, err
 	}
-	db, err := sql.Open(driverName, dsn)
 	if err != nil {
 		return nil, err
 	}
 	return db, nil
 }
 
 func createDatabase(cfg *Config) error {
+	// use default database to create new database
 	if cfg.Database == defaultDatabase {
 		return nil
 	}
-	// use default database to create new database
```

Comment:
it is nessary to switch to default database to create new database.
if the database not default and not exist, `cfg.buildDB` will return no such database error.

so last PR #18067 split `dsn` to `endpoint` `database` and other connection_params, make it easy to switch default database, replace is fragile (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16476).

Reply:
Hi, I see your point, but in that case, we don't manipulate strings. We overwrite the database field after parsing all options. The database name from the DSN URLs is ignored after all.

Reply:
LGTM, @dmitryax @StarpTech  please take a look.